File: src/test/java/com/weeth/domain/comment/test/fixture/CommentTestFixture.java
package com.weeth.domain.comment.test.fixture;

import com.weeth.domain.board.domain.entity.Notice;
import com.weeth.domain.comment.domain.entity.Comment;
import com.weeth.domain.user.domain.entity.User;

import java.util.ArrayList;

public class CommentTestFixture {
    public static Comment createComment(Long id, String content, User user, Notice noice){
        return Comment.builder()
                .id(id)
                .content(content)
                .notice(noice)
                .user(user)
                .children(new ArrayList<>())
                .isDeleted(Boolean.FALSE)
                .build();
    }
}

File: src/test/kotlin/com/weeth/domain/comment/fixture/CommentTestFixture.kt
package com.weeth.domain.comment.fixture

import com.weeth.domain.board.domain.entity.Post
import com.weeth.domain.club.domain.entity.ClubMember
import com.weeth.domain.club.fixture.ClubMemberTestFixture
import com.weeth.domain.comment.domain.entity.Comment

object CommentTestFixture {
    fun createPostComment(
        id: Long = 1L,
        content: String = "테스트 댓글",
        post: Post,
        clubMember: ClubMember = ClubMemberTestFixture.createActiveMember(),
        parent: Comment? = null,
        isDeleted: Boolean = false,
    ) = Comment(
        id = id,
        content = content,
        post = post,
        clubMember = clubMember,
        parent = parent,
        isDeleted = isDeleted,
    )
}
